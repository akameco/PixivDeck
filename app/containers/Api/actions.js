// @flow
import type { Response } from 'services/api'
import type { Action } from './actionTypes'
import { API_REQUEST_SUCCESS } from './constants'

export function apiRequestSuccess(response: Response): Action {
  return {
    type: API_REQUEST_SUCCESS,
    response,
  }
}
