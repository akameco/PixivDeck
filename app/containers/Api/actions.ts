import { Response } from 'services/api'
import { Action } from './actionTypes'
import { API_REQUEST_SUCCESS } from './constants'

export function apiRequestSuccess(response: Response): Action {
  return {
    type: API_REQUEST_SUCCESS,
    response,
  }
}
