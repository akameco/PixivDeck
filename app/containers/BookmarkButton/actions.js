// @flow
import type { Action } from './actionTypes'
import {
  ADD_BOOKMARK_REQUEST,
  ADD_BOOKMARK_SUCCESS,
  DELETE_BOOKMARK_REQUEST,
  DELETE_BOOKMARK_SUCCESS,
  ADD_BOOKMARK_FAILER,
  DELETE_BOOKMARK_FAILER,
} from './constants'
import type { Restrict } from './types'

export function addBookmarkRequest(id: number, restrict: Restrict): Action {
  return {
    type: ADD_BOOKMARK_REQUEST,
    id,
    restrict,
  }
}

export function addBookmarkSuccess(id: number, restrict: Restrict): Action {
  return {
    type: ADD_BOOKMARK_SUCCESS,
    id,
    restrict,
  }
}

export function deleteBookmarkRequest(id: number): Action {
  return {
    type: DELETE_BOOKMARK_REQUEST,
    id,
  }
}

export function deleteBookmarkSuccess(id: number): Action {
  return {
    type: DELETE_BOOKMARK_SUCCESS,
    id,
  }
}

export function addBookmarkFailer(id: number, error: string): Action {
  return {
    type: ADD_BOOKMARK_FAILER,
    id,
    error,
  }
}

export function deleteBookmarkFailer(id: number, error: string): Action {
  return {
    type: DELETE_BOOKMARK_FAILER,
    id,
    error,
  }
}
