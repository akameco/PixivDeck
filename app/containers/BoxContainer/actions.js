// @flow
import type { Action } from './actionTypes'
import { OPEN_PIXIV } from './constants'

export function openPixiv(id: number): Action {
  return {
    type: OPEN_PIXIV,
    id,
  }
}
