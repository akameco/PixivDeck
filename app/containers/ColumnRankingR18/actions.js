// @flow
import type { Action } from './actionTypes'
import {
  ADD_COLUMN,
  ADD_COLUMN_SUCCESS,
  FETCH_NEXT,
  FETCH_NEXT_FAILRE,
  FETCH,
  FETCH_FAILRE,
  SET_NEXT_URL,
  FETCH_SUCCESS,
  FETCH_NEXT_SUCCESS,
} from './constants'
import type { ColumnId } from './reducer'

export function addColumn(id: ColumnId): Action {
  return {
    type: ADD_COLUMN,
    id,
  }
}

export function addColumnSuccess(id: ColumnId): Action {
  return {
    type: ADD_COLUMN_SUCCESS,
    id,
  }
}

export function fetchNext(id: ColumnId): Action {
  return {
    type: FETCH_NEXT,
    id,
  }
}

export function fetchNextFailre(id: ColumnId): Action {
  return {
    type: FETCH_NEXT_FAILRE,
    id,
  }
}

export function fetch(id: ColumnId): Action {
  return {
    type: FETCH,
    id,
  }
}

export function fetchFailre(id: ColumnId): Action {
  return {
    type: FETCH_FAILRE,
    id,
  }
}

export function setNextUrl(id: ColumnId, nextUrl: string): Action {
  return {
    type: SET_NEXT_URL,
    id,
    nextUrl,
  }
}

export function fetchSuccess(id: ColumnId, ids: Array<number>): Action {
  return {
    type: FETCH_SUCCESS,
    id,
    ids,
  }
}

export function fetchNextSuccess(id: ColumnId, ids: Array<number>): Action {
  return {
    type: FETCH_NEXT_SUCCESS,
    id,
    ids,
  }
}
