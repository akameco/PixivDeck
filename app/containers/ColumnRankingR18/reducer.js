// @flow
import ms from 'ms'
import update from 'utils/update'
import type { Action } from './actionTypes'
import * as Actions from './constants'
import { baseReducer, type BaseColumn } from '../Column/reducer'

export type R18Mode =
  | 'day_r18'
  | 'week_r18'
  | 'day_male_r18'
  | 'day_female_r18'
  | 'week_r18g'

export type ColumnId = R18Mode

export type ColumnRankingR18 = { interval: number } & BaseColumn

export type State = $Shape<{ [Mode]: ColumnRankingR18 }>

export const initialState: State = {}

export default function(state: State = initialState, action: Action): State {
  switch (action.type) {
    case Actions.ADD_COLUMN_SUCCESS:
      return update(state, action, {
        ids: [],
        nextUrl: null,
        interval: ms('3h'),
      })
    case Actions.CLERE:
      return update(state, action, {
        ids: [],
        nextUrl: null,
        interval: ms('3h'),
      })
    default:
      return baseReducer('ColumnRankingR18', Actions, state, action)
  }
}
