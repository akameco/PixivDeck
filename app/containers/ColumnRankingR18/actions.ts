import { Action } from './actionTypes'
import {
  ADD_COLUMN,
  ADD_COLUMN_SUCCESS,
  CLERE,
  FETCH_NEXT,
  FETCH_NEXT_FAILRE,
  FETCH,
  FETCH_FAILRE,
  WATCH_NEW,
  START_WATCH,
  SET_NEXT_URL,
  FETCH_SUCCESS,
  FETCH_NEXT_SUCCESS,
} from './constants'
import { ColumnId } from './reducer'

export function addColumn(id: ColumnId): Action {
  return {
    type: ADD_COLUMN,
    id,
  }
}
export function addColumnSuccess(id: ColumnId): Action {
  return {
    type: ADD_COLUMN_SUCCESS,
    id,
  }
}
export function clere(id: ColumnId): Action {
  return {
    type: CLERE,
    id,
  }
}
export function fetchNext(id: ColumnId): Action {
  return {
    type: FETCH_NEXT,
    id,
  }
}
export function fetchNextFailre(id: ColumnId): Action {
  return {
    type: FETCH_NEXT_FAILRE,
    id,
  }
}
export function fetch(id: ColumnId): Action {
  return {
    type: FETCH,
    id,
  }
}
export function fetchFailre(id: ColumnId): Action {
  return {
    type: FETCH_FAILRE,
    id,
  }
}
export function watchNew(id: ColumnId): Action {
  return {
    type: WATCH_NEW,
    id,
  }
}
export function startWatch(id: ColumnId): Action {
  return {
    type: START_WATCH,
    id,
  }
}
export function setNextUrl(id: ColumnId, nextUrl: string): Action {
  return {
    type: SET_NEXT_URL,
    id,
    nextUrl,
  }
}
export function fetchSuccess(id: ColumnId, ids: number[]): Action {
  return {
    type: FETCH_SUCCESS,
    id,
    ids,
  }
}
export function fetchNextSuccess(id: ColumnId, ids: number[]): Action {
  return {
    type: FETCH_NEXT_SUCCESS,
    id,
    ids,
  }
}
