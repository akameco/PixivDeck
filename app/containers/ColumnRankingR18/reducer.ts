import ms from 'ms'
import update from 'utils/update'
import { baseReducer, BaseColumn } from '../Column/reducer'
import { Action } from './actionTypes'
import * as Actions from './constants'

export type R18Mode =
  | 'day_r18'
  | 'week_r18'
  | 'day_male_r18'
  | 'day_female_r18'
  | 'week_r18g'
export type ColumnId = R18Mode
export type ColumnRankingR18 = {
  interval: number
} & BaseColumn
export interface State {
  [a: string]: Partial<ColumnRankingR18>
}
export const initialState: State = {}
export default function(state: State = initialState, action: Action): State {
  switch (action.type) {
    case Actions.ADD_COLUMN_SUCCESS:
    case Actions.CLERE:
      return update(state, action, {
        ids: [],
        nextUrl: null,
        interval: ms('3h'),
      })

    default:
      return baseReducer('ColumnRankingR18', Actions, state, action)
  }
}
