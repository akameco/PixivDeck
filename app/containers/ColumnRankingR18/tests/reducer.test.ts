import reducer from '../reducer'
import * as actions from '../actions'

test('default action', () => {
  expect(
    reducer(undefined, {
      // @ts-ignore
      type: 'default action',
    })
  ).toMatchSnapshot()
})

test('ADD_COLUMN_SUCCESS', () => {
  expect(
    reducer({}, actions.addColumnSuccess('day_male_r18'))
  ).toMatchSnapshot()
})

test('SET_NEXT_URL', () => {
  expect(
    reducer({}, actions.setNextUrl('day_r18', 'https://nextUrl1'))
  ).toMatchSnapshot()
  expect(
    reducer({}, actions.setNextUrl('day_r18', 'https://nextUrl/2'))
  ).toMatchSnapshot()
})
