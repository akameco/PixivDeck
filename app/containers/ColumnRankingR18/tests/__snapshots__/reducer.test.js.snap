// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADD_COLUMN_SUCCESS 1`] = `
Object {
  "day_male_r18": Object {
    "ids": Array [],
    "interval": 10800000,
    "nextUrl": null,
  },
}
`;

exports[`SET_NEXT_URL 1`] = `
Object {
  "day_r18": Object {
    "nextUrl": "https://nextUrl1",
  },
}
`;

exports[`SET_NEXT_URL 2`] = `
Object {
  "day_r18": Object {
    "nextUrl": "https://nextUrl/2",
  },
}
`;

exports[`default action 1`] = `Object {}`;
